src/pynwb/file.py

```diff
@@ -589,7 +589,11 @@ def add_epoch(self, **kwargs):
         """
         self.__check_epochs()
         if kwargs['tags'] is not None:
-            self.epoch_tags.update(kwargs['tags'])
+            # If a str is passed into epoch_tags directly, it gets split into characters
+            tmp = kwargs['tags']
+            if isinstance(kwargs['tags'], str):
+                tmp = [s.strip() for s in kwargs['tags'].split(",") if not s.isspace()]
```

Comment:
If I understand this correctly, then if `tags='tag1, tag2, tag3'` this would split the string to make a list `['tag1'. 'tag2', 'tag3']. It's not clear to me what the use case for allowing implicit definitions of lists via strings is. I agree that the current behavior of splitting a string into characters is not useful, but I think that is largely a bug because the string is treated like a list. Instead, I would suggest treating the data types for what they are. That is, if the user passes a string we should treat it as a single tag, and if the user passes a list we should treat it as a list of tags. Long-story-short, I would suggest changing the code as follows: 


```suggestion
            if isinstance(kwargs['tags'], str):
                tmp = [kwargs['tags'],]  
```

Reply:
This makes sense to me in terms of a clear / consistent behaviour. What I currently wrote is based on making sure `epoch_tags`, which I understand to keep a list of all observed tags is consistent with the tags that get added to the epochs table. To keep this consistent, the strings should be processed in the same way in `pynwb.epoch` line 47. This would overall update the behavior of string tags for anyone currently doing this (the bug here is only about `epoch_tags`, and if anyone was currently using a `tag1, tag2` strategy, these tags are currently split into a list and added to the epochs table.

Reply:
That makes sense. Thanks for the pointer to https://github.com/NeurodataWithoutBorders/pynwb/blob/ed9eb6d16e5031cb1d19f74272364394f1ab3188/src/pynwb/epoch.py#L46-L47 I was not aware of this behavior. In this case, what you have makes perfect sense. 

I would suggest to updated the doc string for tags here:

https://github.com/NeurodataWithoutBorders/pynwb/blob/ed9eb6d16e5031cb1d19f74272364394f1ab3188/src/pynwb/epoch.py#L36-L37

to describe how strings are being treated. I.e., describe that if a string is passed in that the string is expected to be a comma-separated list and will be parsed accordingly. Currently, I don't think this behavior is clear from the docs. Could you add this change on this PR. 

We could consider consider changing how strings are being treated here (i.e., either always require lists or treat strings as a single tag) but I would recommend to do this as a separate issue. For this PR, making sure that the behavior is consistent seems the right thing to do. Thanks for your help!

Reply:
This sounds good / makes sense to me! I have updated the docstring in `pynwb.src.pynwb.epoch` to describe the behaviour for how it treats strings. I also added a note in `file` that processing should match what is done in `epoch`.

I think that should address what's needed for this PR, to fix the bug and update the docs - and then if the behaviour is to be changed at some point, that could be a separate PR.